### PR TITLE
edX Upgrade Koa master

### PR DESCRIPTION
**Pull request by asadali145**
This contains changes from open-release/koa.master, There are a few extra commits but these are due to different persons making the same changes in different branches.

### CODE REVIEW

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/867/workflows/workflows_id/jobs/867/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/open-release/koa.master...HEAD, staged and unstaged changes
-------------
openedx/adg/common/course_meta/admin.py (100%)
openedx/adg/common/course_meta/apps.py (0.0%): Missing lines 4,7-8
openedx/adg/common/course_meta/helpers.py (100%)
openedx/adg/common/course_meta/models.py (94.4%): Missing lines 45
openedx/adg/common/course_meta/tests/factories.py (100%)
openedx/adg/common/course_meta/tests/test_helpers.py (100%)
openedx/adg/common/installed_apps.py (100%)
openedx/adg/common/lib/mandrill_client/client.py (63.6%): Missing lines 25,37-38,43-45,48-49
openedx/adg/common/lib/mandrill_client/email_data.py (37.5%): Missing lines 14-15,17,24-25
openedx/adg/common/mailchimp_pipeline/__init__.py (100%)
openedx/adg/common/mailchimp_pipeline/apps.py (100%)
openedx/adg/common/mailchimp_pipeline/client.py (100%)
openedx/adg/common/mailchimp_pipeline/handlers.py (92.5%): Missing lines 96-98
openedx/adg/common/mailchimp_pipeline/helpers.py (94.7%): Missing lines 22
openedx/adg/common/mailchimp_pipeline/tasks.py (100%)
openedx/adg/common/mailchimp_pipeline/tests/test_client.py (100%)
openedx/adg/common/mailchimp_pipeline/tests/test_handlers.py (100%)
openedx/adg/common/mailchimp_pipeline/tests/test_helpers.py (100%)
openedx/adg/common/mailchimp_pipeline/tests/test_tasks.py (100%)
openedx/adg/context_processors.py (0.0%): Missing lines 4,7,11
openedx/adg/lms/applications/__init__.py (100%)
openedx/adg/lms/applications/admin.py (96.9%): Missing lines 35
openedx/adg/lms/applications/api_urls.py (100%)
openedx/adg/lms/applications/api_views.py (100%)
openedx/adg/lms/applications/apps.py (100%)
openedx/adg/lms/applications/constants.py (100%)
openedx/adg/lms/applications/forms.py (91.4%): Missing lines 106-110
openedx/adg/lms/applications/handlers.py (100%)
openedx/adg/lms/applications/helpers.py (100%)
openedx/adg/lms/applications/management/commands/tests/test_update_is_prerequisite_courses_passed.py (100%)
openedx/adg/lms/applications/management/commands/update_is_prerequisite_courses_passed.py (100%)
openedx/adg/lms/applications/models.py (95.5%): Missing lines 80,99,210,226,243
openedx/adg/lms/applications/serializers.py (100%)
openedx/adg/lms/applications/test/constants.py (100%)
openedx/adg/lms/applications/test/factories.py (100%)
openedx/adg/lms/applications/test/test_api_views.py (100%)
openedx/adg/lms/applications/test/test_forms.py (100%)
openedx/adg/lms/applications/test/test_handlers.py (100%)
openedx/adg/lms/applications/test/test_helpers.py (100%)
openedx/adg/lms/applications/test/test_models.py (100%)
openedx/adg/lms/applications/test/test_serializers.py (100%)
openedx/adg/lms/applications/test/test_views.py (100%)
openedx/adg/lms/applications/urls.py (100%)
openedx/adg/lms/applications/views.py (68.9%): Missing lines 185,195-196,198-204,206-209,218,228,230-233,235-238,240,253-257,259-260,265
openedx/adg/lms/branding_extension/helpers.py (100%)
openedx/adg/lms/branding_extension/tests/helpers_tests.py (100%)
openedx/adg/lms/registration_extension/admin.py (100%)
openedx/adg/lms/registration_extension/apps.py (0.0%): Missing lines 4,7-8
openedx/adg/lms/registration_extension/constants.py (100%)
openedx/adg/lms/registration_extension/forms.py (68.8%): Missing lines 36-37,58-59,61,86,88,94,119,130
openedx/adg/lms/registration_extension/models.py (100%)
openedx/adg/lms/registration_extension/tests/factories.py (100%)
openedx/adg/lms/student/helpers.py (100%)
openedx/adg/lms/student/tests/test_helpers.py (100%)
openedx/adg/lms/urls.py (100%)
openedx/adg/lms/user_authn_override/login_form.py (100%)
openedx/adg/lms/user_authn_override/password_reset.py (100%)
openedx/adg/lms/user_authn_override/tests/test_login_form.py (100%)
openedx/adg/lms/user_authn_override/tests/test_password_reset.py (100%)
openedx/adg/lms/utils/date_utils.py (100%)
openedx/adg/lms/utils/decorators.py (100%)
openedx/adg/lms/utils/env_utils.py (100%)
openedx/adg/lms/utils/tests/test_env_utils.py (100%)
openedx/adg/lms/utils/tests/tests_date_utils.py (100%)
-------------
Total:   1740 lines
Missing: 81 lines
Coverage: 95%
-------------
```

**Comment by asadali145** [comment]
Changing base branch to check if all the tests of edx are passing.

**Comment by asadali145** [comment]
All the tests were successful but diff coverage report was not generated due to too many changes against master.

**Review by annas-usman — APPROVED** [review]
I verified the core changes assigned to me, everything looks in accordance to the core changes mentioned in wiki.

File: lms/urls.py [hunk 1]
```diff
@@ -676,6 +677,9 @@
     ),
 ]
 
+# include all adg lms urls
```
Comment (ahmed-faraz46) [inline lms/urls.py:680]: [Core change 1.1 ](https://github.com/OmnipreneurshipAcademy/edx-platform/wiki/Core-Changes#11-write-required-apis-for-application-page-2) is not moved here. I guess that core change is not valid, would please have a look and update the wiki accordingly.
@asadali145

File: lms/envs/test.py [hunk 2]
```diff
@@ -591,3 +591,92 @@
 LOGISTRATION_API_RATELIMIT = '5/m'
 
 REGISTRATION_VALIDATION_RATELIMIT = '5/minute'
+
+############### ADG #####################
```
Comment (ahmed-faraz46) [inline lms/envs/test.py:595]: Let's please update the core change 12 it is not reflecting all of these changes.
```
Added All languages list in lms/envs/test.py and cms/envs/test.py. (I have added a direct commit to koa-master branch and cms/envs/test.py was updated in that commit.)
lms/envs/test.py Added LANGUAGES = [list of all languages]

cms/envs/test.py Added LANGUAGES = [list of all languages]. Added under this commit.

Tests were failing due to updated languages list. We changed languages to make changes for transifex supported langs.

Can override using decorator in tests but this was used at multiple places so added it in test.py.
```

File: lms/djangoapps/commerce/api/v1/tests/test_views.py [hunk 3]
```diff
@@ -18,6 +18,7 @@
 from common.djangoapps.course_modes.models import CourseMode
 from lms.djangoapps.verify_student.models import VerificationDeadline
 from common.djangoapps.student.tests.factories import UserFactory
+from openedx.adg.lms.applications.constants import MAXIMUM_YEAR_OPTION
```
Comment (ahmed-faraz46) [inline lms/djangoapps/commerce/api/v1/tests/test_views.py:21]: Let's please just back port whatever edX has done. Also the core change in the wiki should be updated accordingly.

**Review by ahmed-faraz46 — CHANGES_REQUESTED** [review]

File: lms/djangoapps/commerce/api/v1/tests/test_views.py (hunk 3 above)
Comment (asadali145) [reply lms/djangoapps/commerce/api/v1/tests/test_views.py:21]: Good catch this is an un-used import. Forgot to remove this after back port.

File: lms/envs/test.py (hunk 2 above)
Comment (asadali145) [reply lms/envs/test.py:595]: These changes were made under different PRs. All other changes made here are listed in core change [8](https://github.com/OmnipreneurshipAcademy/edx-platform/wiki/Core-Changes#8-mailchimp-integration-pr-lp-2447).

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/913/workflows/workflows_id/jobs/913/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/open-release/koa.master...HEAD, staged and unstaged changes
-------------
openedx/adg/common/course_meta/admin.py (100%)
openedx/adg/common/course_meta/apps.py (0.0%): Missing lines 4,7-8
openedx/adg/common/course_meta/helpers.py (100%)
openedx/adg/common/course_meta/models.py (94.4%): Missing lines 45
openedx/adg/common/course_meta/tests/factories.py (100%)
openedx/adg/common/course_meta/tests/test_helpers.py (100%)
openedx/adg/common/installed_apps.py (100%)
openedx/adg/common/lib/mandrill_client/client.py (63.6%): Missing lines 25,37-38,43-45,48-49
openedx/adg/common/lib/mandrill_client/email_data.py (37.5%): Missing lines 14-15,17,24-25
openedx/adg/common/mailchimp_pipeline/__init__.py (100%)
openedx/adg/common/mailchimp_pipeline/apps.py (100%)
openedx/adg/common/mailchimp_pipeline/client.py (100%)
openedx/adg/common/mailchimp_pipeline/handlers.py (92.5%): Missing lines 96-98
openedx/adg/common/mailchimp_pipeline/helpers.py (94.7%): Missing lines 22
openedx/adg/common/mailchimp_pipeline/tasks.py (100%)
openedx/adg/common/mailchimp_pipeline/tests/test_client.py (100%)
openedx/adg/common/mailchimp_pipeline/tests/test_handlers.py (100%)
openedx/adg/common/mailchimp_pipeline/tests/test_helpers.py (100%)
openedx/adg/common/mailchimp_pipeline/tests/test_tasks.py (100%)
openedx/adg/context_processors.py (0.0%): Missing lines 4,7,11
openedx/adg/lms/applications/__init__.py (100%)
openedx/adg/lms/applications/admin.py (96.9%): Missing lines 35
openedx/adg/lms/applications/api_urls.py (100%)
openedx/adg/lms/applications/api_views.py (100%)
openedx/adg/lms/applications/apps.py (100%)
openedx/adg/lms/applications/constants.py (100%)
openedx/adg/lms/applications/forms.py (91.4%): Missing lines 106-110
openedx/adg/lms/applications/handlers.py (100%)
openedx/adg/lms/applications/helpers.py (100%)
openedx/adg/lms/applications/management/commands/tests/test_update_is_prerequisite_courses_passed.py (100%)
openedx/adg/lms/applications/management/commands/update_is_prerequisite_courses_passed.py (100%)
openedx/adg/lms/applications/models.py (95.5%): Missing lines 80,99,210,226,243
openedx/adg/lms/applications/serializers.py (100%)
openedx/adg/lms/applications/test/constants.py (100%)
openedx/adg/lms/applications/test/factories.py (100%)
openedx/adg/lms/applications/test/test_api_views.py (100%)
openedx/adg/lms/applications/test/test_forms.py (100%)
openedx/adg/lms/applications/test/test_handlers.py (100%)
openedx/adg/lms/applications/test/test_helpers.py (100%)
openedx/adg/lms/applications/test/test_models.py (100%)
openedx/adg/lms/applications/test/test_serializers.py (100%)
openedx/adg/lms/applications/test/test_views.py (100%)
openedx/adg/lms/applications/urls.py (100%)
openedx/adg/lms/applications/views.py (68.9%): Missing lines 185,195-196,198-204,206-209,218,228,230-233,235-238,240,253-257,259-260,265
openedx/adg/lms/branding_extension/helpers.py (100%)
openedx/adg/lms/branding_extension/tests/helpers_tests.py (100%)
openedx/adg/lms/registration_extension/admin.py (100%)
openedx/adg/lms/registration_extension/apps.py (0.0%): Missing lines 4,7-8
openedx/adg/lms/registration_extension/constants.py (100%)
openedx/adg/lms/registration_extension/forms.py (68.8%): Missing lines 36-37,58-59,61,86,88,94,119,130
openedx/adg/lms/registration_extension/models.py (100%)
openedx/adg/lms/registration_extension/tests/factories.py (100%)
openedx/adg/lms/student/helpers.py (100%)
openedx/adg/lms/student/tests/test_helpers.py (100%)
openedx/adg/lms/urls.py (100%)
openedx/adg/lms/user_authn_override/login_form.py (100%)
openedx/adg/lms/user_authn_override/password_reset.py (100%)
openedx/adg/lms/user_authn_override/tests/test_login_form.py (100%)
openedx/adg/lms/user_authn_override/tests/test_password_reset.py (100%)
openedx/adg/lms/utils/date_utils.py (100%)
openedx/adg/lms/utils/decorators.py (100%)
openedx/adg/lms/utils/env_utils.py (100%)
openedx/adg/lms/utils/tests/test_env_utils.py (100%)
openedx/adg/lms/utils/tests/tests_date_utils.py (100%)
-------------
Total:   1740 lines
Missing: 81 lines
Coverage: 95%
-------------
```

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/920/workflows/workflows_id/jobs/920/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/open-release/koa.master...HEAD, staged and unstaged changes
-------------
openedx/adg/common/course_meta/admin.py (100%)
openedx/adg/common/course_meta/apps.py (0.0%): Missing lines 4,7-8
openedx/adg/common/course_meta/helpers.py (100%)
openedx/adg/common/course_meta/models.py (94.4%): Missing lines 45
openedx/adg/common/course_meta/tests/factories.py (100%)
openedx/adg/common/course_meta/tests/test_helpers.py (100%)
openedx/adg/common/installed_apps.py (100%)
openedx/adg/common/lib/mandrill_client/client.py (63.6%): Missing lines 25,37-38,43-45,48-49
openedx/adg/common/lib/mandrill_client/email_data.py (37.5%): Missing lines 14-15,17,24-25
openedx/adg/common/mailchimp_pipeline/__init__.py (100%)
openedx/adg/common/mailchimp_pipeline/apps.py (100%)
openedx/adg/common/mailchimp_pipeline/client.py (100%)
openedx/adg/common/mailchimp_pipeline/handlers.py (92.5%): Missing lines 96-98
openedx/adg/common/mailchimp_pipeline/helpers.py (94.7%): Missing lines 22
openedx/adg/common/mailchimp_pipeline/tasks.py (100%)
openedx/adg/common/mailchimp_pipeline/tests/test_client.py (100%)
openedx/adg/common/mailchimp_pipeline/tests/test_handlers.py (100%)
openedx/adg/common/mailchimp_pipeline/tests/test_helpers.py (100%)
openedx/adg/common/mailchimp_pipeline/tests/test_tasks.py (100%)
openedx/adg/context_processors.py (0.0%): Missing lines 4,7,11
openedx/adg/lms/applications/__init__.py (100%)
openedx/adg/lms/applications/admin.py (96.9%): Missing lines 35
openedx/adg/lms/applications/api_urls.py (100%)
openedx/adg/lms/applications/api_views.py (100%)
openedx/adg/lms/applications/apps.py (100%)
openedx/adg/lms/applications/constants.py (100%)
openedx/adg/lms/applications/forms.py (91.4%): Missing lines 106-110
openedx/adg/lms/applications/handlers.py (100%)
openedx/adg/lms/applications/helpers.py (100%)
openedx/adg/lms/applications/management/commands/tests/test_update_is_prerequisite_courses_passed.py (100%)
openedx/adg/lms/applications/management/commands/update_is_prerequisite_courses_passed.py (100%)
openedx/adg/lms/applications/models.py (95.5%): Missing lines 80,99,210,226,243
openedx/adg/lms/applications/serializers.py (100%)
openedx/adg/lms/applications/test/constants.py (100%)
openedx/adg/lms/applications/test/factories.py (100%)
openedx/adg/lms/applications/test/test_api_views.py (100%)
openedx/adg/lms/applications/test/test_forms.py (100%)
openedx/adg/lms/applications/test/test_handlers.py (100%)
openedx/adg/lms/applications/test/test_helpers.py (100%)
openedx/adg/lms/applications/test/test_models.py (100%)
openedx/adg/lms/applications/test/test_serializers.py (100%)
openedx/adg/lms/applications/test/test_views.py (100%)
openedx/adg/lms/applications/urls.py (100%)
openedx/adg/lms/applications/views.py (68.9%): Missing lines 185,195-196,198-204,206-209,218,228,230-233,235-238,240,253-257,259-260,265
openedx/adg/lms/branding_extension/helpers.py (100%)
openedx/adg/lms/branding_extension/tests/helpers_tests.py (100%)
openedx/adg/lms/registration_extension/admin.py (100%)
openedx/adg/lms/registration_extension/apps.py (0.0%): Missing lines 4,7-8
openedx/adg/lms/registration_extension/constants.py (100%)
openedx/adg/lms/registration_extension/forms.py (68.8%): Missing lines 36-37,58-59,61,86,88,94,119,130
openedx/adg/lms/registration_extension/models.py (100%)
openedx/adg/lms/registration_extension/tests/factories.py (100%)
openedx/adg/lms/student/helpers.py (100%)
openedx/adg/lms/student/tests/test_helpers.py (100%)
openedx/adg/lms/urls.py (100%)
openedx/adg/lms/user_authn_override/login_form.py (100%)
openedx/adg/lms/user_authn_override/password_reset.py (100%)
openedx/adg/lms/user_authn_override/tests/test_login_form.py (100%)
openedx/adg/lms/user_authn_override/tests/test_password_reset.py (100%)
openedx/adg/lms/utils/date_utils.py (100%)
openedx/adg/lms/utils/decorators.py (100%)
openedx/adg/lms/utils/env_utils.py (100%)
openedx/adg/lms/utils/tests/test_env_utils.py (100%)
openedx/adg/lms/utils/tests/tests_date_utils.py (100%)
-------------
Total:   1740 lines
Missing: 81 lines
Coverage: 95%
-------------
```

**Comment by zahid-arbisoft** [comment]
## Diff Coverage Report - ADG
Check-out complete **[report](https://app.circleci.com/pipelines/github/OmnipreneurshipAcademy/edx-platform/921/workflows/workflows_id/jobs/921/artifacts)**. Coverage !!! :rocket:
```
-------------
Diff Coverage
Diff: origin/open-release/koa.master...HEAD, staged and unstaged changes
-------------
openedx/adg/common/course_meta/admin.py (100%)
openedx/adg/common/course_meta/apps.py (0.0%): Missing lines 4,7-8
openedx/adg/common/course_meta/helpers.py (100%)
openedx/adg/common/course_meta/models.py (94.4%): Missing lines 45
openedx/adg/common/course_meta/tests/factories.py (100%)
openedx/adg/common/course_meta/tests/test_helpers.py (100%)
openedx/adg/common/installed_apps.py (100%)
openedx/adg/common/lib/mandrill_client/client.py (63.6%): Missing lines 25,37-38,43-45,48-49
openedx/adg/common/lib/mandrill_client/email_data.py (37.5%): Missing lines 14-15,17,24-25
openedx/adg/common/mailchimp_pipeline/__init__.py (100%)
openedx/adg/common/mailchimp_pipeline/apps.py (100%)
openedx/adg/common/mailchimp_pipeline/client.py (100%)
openedx/adg/common/mailchimp_pipeline/handlers.py (92.5%): Missing lines 96-98
openedx/adg/common/mailchimp_pipeline/helpers.py (94.7%): Missing lines 22
openedx/adg/common/mailchimp_pipeline/tasks.py (100%)
openedx/adg/common/mailchimp_pipeline/tests/test_client.py (100%)
openedx/adg/common/mailchimp_pipeline/tests/test_handlers.py (100%)
openedx/adg/common/mailchimp_pipeline/tests/test_helpers.py (100%)
openedx/adg/common/mailchimp_pipeline/tests/test_tasks.py (100%)
openedx/adg/context_processors.py (0.0%): Missing lines 4,7,11
openedx/adg/lms/applications/__init__.py (100%)
openedx/adg/lms/applications/admin.py (96.9%): Missing lines 35
openedx/adg/lms/applications/api_urls.py (100%)
openedx/adg/lms/applications/api_views.py (100%)
openedx/adg/lms/applications/apps.py (100%)
openedx/adg/lms/applications/constants.py (100%)
openedx/adg/lms/applications/forms.py (91.4%): Missing lines 106-110
openedx/adg/lms/applications/handlers.py (100%)
openedx/adg/lms/applications/helpers.py (100%)
openedx/adg/lms/applications/management/commands/tests/test_update_is_prerequisite_courses_passed.py (100%)
openedx/adg/lms/applications/management/commands/update_is_prerequisite_courses_passed.py (100%)
openedx/adg/lms/applications/models.py (95.5%): Missing lines 80,99,210,226,243
openedx/adg/lms/applications/serializers.py (100%)
openedx/adg/lms/applications/test/constants.py (100%)
openedx/adg/lms/applications/test/factories.py (100%)
openedx/adg/lms/applications/test/test_api_views.py (100%)
openedx/adg/lms/applications/test/test_forms.py (100%)
openedx/adg/lms/applications/test/test_handlers.py (100%)
openedx/adg/lms/applications/test/test_helpers.py (100%)
openedx/adg/lms/applications/test/test_models.py (100%)
openedx/adg/lms/applications/test/test_serializers.py (100%)
openedx/adg/lms/applications/test/test_views.py (100%)
openedx/adg/lms/applications/urls.py (100%)
openedx/adg/lms/applications/views.py (68.9%): Missing lines 185,195-196,198-204,206-209,218,228,230-233,235-238,240,253-257,259-260,265
openedx/adg/lms/branding_extension/helpers.py (100%)
openedx/adg/lms/branding_extension/tests/helpers_tests.py (100%)
openedx/adg/lms/registration_extension/admin.py (100%)
openedx/adg/lms/registration_extension/apps.py (0.0%): Missing lines 4,7-8
openedx/adg/lms/registration_extension/constants.py (100%)
openedx/adg/lms/registration_extension/forms.py (68.8%): Missing lines 36-37,58-59,61,86,88,94,119,130
openedx/adg/lms/registration_extension/models.py (100%)
openedx/adg/lms/registration_extension/tests/factories.py (100%)
openedx/adg/lms/student/helpers.py (100%)
openedx/adg/lms/student/tests/test_helpers.py (100%)
openedx/adg/lms/urls.py (100%)
openedx/adg/lms/user_authn_override/login_form.py (100%)
openedx/adg/lms/user_authn_override/password_reset.py (100%)
openedx/adg/lms/user_authn_override/tests/test_login_form.py (100%)
openedx/adg/lms/user_authn_override/tests/test_password_reset.py (100%)
openedx/adg/lms/utils/date_utils.py (100%)
openedx/adg/lms/utils/decorators.py (100%)
openedx/adg/lms/utils/env_utils.py (100%)
openedx/adg/lms/utils/tests/test_env_utils.py (100%)
openedx/adg/lms/utils/tests/tests_date_utils.py (100%)
-------------
Total:   1740 lines
Missing: 81 lines
Coverage: 95%
-------------
```

**Comment by asadali145** [comment]
As we have tested the koa release on our servers, I am closing this one.

**Pull request closed without merging**